File: webPart/src/main/web/src/app/models/menu.ts
import { IngredientUsed } from "./ingredientUsed";
import { Jour } from "./jour";
import { User } from "./user";


export class Menu {
  id: number;
  user: User;
  numeroSemaine: number;
  semaine: Jour[];
  ingredientsUsed: IngredientUsed[];


  constructor(numSem){
    this.numeroSemaine = numSem
    this.semaine = []
    this.semaine.push(new Jour("Lundi"))
    this.semaine.push(new Jour("Mardi"))
    this.semaine.push(new Jour("Mercredi"))
    this.semaine.push(new Jour("Jeudi"))
    this.semaine.push(new Jour("Vendredi"))
    this.semaine.push(new Jour("Samedi"))
    this.semaine.push(new Jour("Dimanche"))

    this.ingredientsUsed = []
  }



}

File: serverPart/src/main/java/com/wild/corp/model/Menu.java
package com.wild.corp.model;


import org.hibernate.annotations.LazyCollection;
import org.hibernate.annotations.LazyCollectionOption;

import javax.persistence.*;
import java.util.*;


@Entity
@Table(name = "MENU")
public class Menu {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "ID", nullable = false)
    private Integer id;

    @Column(name = "NUMERO_SEMAINE")
    private Integer numeroSemaine;

    @LazyCollection(LazyCollectionOption.FALSE)
    @OneToMany(mappedBy="menu" , orphanRemoval=true)
    private Set<Jour> semaine;

    @LazyCollection(LazyCollectionOption.FALSE)
    @OneToMany(mappedBy="menu")
    private Set<IngredientUsed> ingredientsUsed;

    @ManyToOne
    private User user;

    void Menu(){
        ingredientsUsed = new HashSet<>();
        semaine = new HashSet<>();
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getNumeroSemaine() {
        return numeroSemaine;
    }

    public void setNumeroSemaine(Integer numeroSemaine) {
        this.numeroSemaine = numeroSemaine;
    }

    public Set<Jour> getSemaine() {
        return semaine;
    }

    public void setSemaine(Set<Jour> semaine) {
        this.semaine = semaine;
    }

    public Set<IngredientUsed> getIngredientsUsed() {
        return ingredientsUsed;
    }

    public void setIngredientsUsed(Set<IngredientUsed> ingredientUsed) {
        this.ingredientsUsed = ingredientsUsed;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Menu)) return false;
        Menu menu = (Menu) o;
        return Objects.equals(getId(), menu.getId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getId());
    }
}
